cfme/common/provider.py

```diff
@@ -337,6 +347,24 @@ def _fill_candu_endpoint_dicts(self, provider_attributes, connection_configs):
             candu_connection["endpoint"]["verify_ssl"] = 0
         connection_configs.append(candu_connection)
 
+    def _fill_rsa_endpoint_dicts(self, provider_attributes, connection_configs):
+        """Fills dicts with rsa endpoint data.
+
+        Helper method for ``self.create_rest``
+        """
+        if "rsa_keypair" not in self.endpoints:
+            return
+
+        endpoint_rsa = self.endpoints["rsa_keypair"]
+        if isinstance(provider_attributes["credentials"], dict):
+            provider_attributes["credentials"] = [provider_attributes["credentials"]]
+
+        provider_attributes["credentials"].append({
+            "userid": endpoint_rsa.credentials.principal,
+            "password": endpoint_rsa.credentials.secret,
```

Comment:
"Credential validation was not successful: Missing credentials"

If I create provider using "get_crud" validation is successful.
Password is located in the credentials.local.yaml
For example:

oc_ci:
  password: ZF9rJxtdUvXEpeJxtJhTZyYd6
  username: admin
uc_ci:
  password: 28f2136dcd0aeb33a54c2974c63421a2313e2f2d
  tenant: admin
  username: admin
uc_ssh_creds:
  username: heat-admin
  password: '-----BEGIN RSA PRIVATE KEY-----

MIIEowIBAAKCAQEAw5b0s96qG4iGd5zDyS9JwD5HWkNcvoF7Mwusk/AY3OSx2Ow6
hQYQH1kWXhfOs6hYWua9mPILiyXmcAKPzWkbauVauZODkTXteVZHTM69nNJn5jbn
BXtARgWYsjQLmILQMmwC0mJvb7pkJ4mlmVa23jqoZZeY88AD7b+SxUB38Xl8iQwI
ROhfpyBAa/42BvVziBbWIdtVgeJRsWwm3TIoc7cdQUCixgus5synL5HsXRla5SQq
anzGd0rlCe6K1hyGupocs6T0F3XzRntqE8lLch1HVtnTdy2dFeotVkOHrCe0Yrt+
BfY27vPlVS4zZrPkqjPBBKtaQMqyfphplRDd+QIDAQABAoIBAEBwg76fJbxchrDx
3S9zfQDxg+ChQPUjA8wVlI5JFGfykQW/Dqi1DqrFVzE49skKNdH9I4o4dMUrf3h3
35j1gCr/1Zw+laJJ+it32ZXcp2v7hm+6tvJ/YPWw8lYD/64Hq0BxhZx7Aev1pPnB
x+V1RFqDSOfvwlzMbn7uFxUV4G6V2/hJFA6W0rRMWxXqh4A6RcisshzKzzOF0E8R
JUavletMWRzFTrdvN1IDhOwevatWYVJ8bqO22JlZLp34Fxy6ge3dzWqEiR/plQwf
0m3gbzaeGro+VUWmvFmWAbKIR5PgfF7vmjwp9+horcLeEnMdMMK9zVE9vFqHFGQb
UuHviAECgYEA9UhdMf/tiRtBY3wnbbETkyafCeiUWJjV1vVK5vWQiKNjoCXPfy2X
SLrDv1SYKHMhTz6sVsAm8Ms753CtP+hJOVXKSMQytRMEuAbDdNbOPqIYjt5L19ty
OYADzIzP7gmn7MhC91C2P+iNDkmae41Bjcqn+roVKP7DqvkR+e8XeSECgYEAzCK+
0K1gH1qnkqR2KT1LkBpNFQ0E1IYn2HvN632IDhcOq5IF6tGsdl5dzxqTlzcIrxEC
3Q3EkXkCF2hsMxCFckk2s34sG97BOdp0ufv5eNufNxdqa3vo5KwpZecFdE84WtE/
ORN7GmA3uKa0wfv57cfBdX2UazVG4E+PFFcgEdkCgYEAwhVFcPfSa+e+IMsvo7M3
CvvHYh6/qy8cHaAwaXorqIKopSLmAfsANxNZKn70eyVGWR79XRNkfKpDt0lIPUY/
EngYvOVKPk9TL45igIQr5BF9Hyc0mhPRWUA7QVK2gOLxFgIMQweBfiYz3vdMnrDS
CzEtI860SaD1EMdOexa92yECgYBxxZOv3lFKaE4o7rxkc6Mfp2j6qIzenf6rKyUz
cH1ZzQQCH5poq3+BYUrLWLaVeri+Sa7c4QWDRnwh67MSKs0VSzTPiSywQ9BRnC2e
Uo9SCqWMIOLREPhWKpXQE5f4oQZ5p1GjT/Y2VxX3NgJcc0FygviRsPH9Zm4gKndG
EODw0QKBgCjblhLJZbHqCZYquG4zLy0tY4V0CwUrC/6HpLjL9ktM+FdS7GwhmZHN
lfugX3DUqfC2YuIUN/x6Yp3KWIqBIuSJzwPDiFOwrK80950x5BvBQVEZifQ325sf
+hljcH0Sgd9Udcs/tTYzjTrgWiKyaEMkZtqvTcHJ9XZnPmqUexZ7

-----END RSA PRIVATE KEY-----'